src/components/CompareResults/SubtestsResults/SubtestsResultsTable.tsx

```diff
@@ -0,0 +1,253 @@
+  columnId: string,
+  uncheckedValues: Set<string>,
+): boolean {
+  const cellConfiguration = cellsConfiguration.find(
+    (cell) => cell.key === columnId,
+  );
+  if (!cellConfiguration || !cellConfiguration.filter) {
+    return true;
+  }
+
+  const { matchesFunction } = cellConfiguration;
+  for (const filterValue of uncheckedValues) {
+    if (matchesFunction(result, filterValue)) {
+      return true;
+    }
+  }
+  return false;
+}
+
+// This function filters the results array using both the searchTerm and the
+// tableFilters. The tableFilters is a map ColumnID -> Set of values to remove.
+function filterResults(
+  results: CompareResultsItem[],
+  searchTerm: string,
+  tableFilters: Map<string, Set<string>>,
+) {
+  if (!searchTerm && !tableFilters.size) {
+    return results;
+  }
+
+  return results.filter((result) => {
+    if (!resultMatchesSearchTerm(result, searchTerm)) {
+      return false;
+    }
+
+    for (const [columnId, uncheckedValues] of tableFilters) {
+      if (resultMatchesColumnFilter(result, columnId, uncheckedValues)) {
+        return false;
+      }
+    }
+
+    return true;
+  });
+}
+
+// const allRevisionsOption =
+//   Strings.components.comparisonRevisionDropdown.allRevisions.key;
+
+type ResultsTableProps = {
+  filteringSearchTerm: string;
+};
+
+function SubtestsResultsTable({ filteringSearchTerm }: ResultsTableProps) {
+  const loaderData = useAsyncValue();
+  const results = loaderData as CompareResultsItem[][];
+  // const activeComparison = useAppSelector(
+  //   (state) => state.comparison.activeComparison,
+  // );
+
+  const [tableFilters, setTableFilters] = useState(
+    new Map() as Map<string, Set<string>>, // ColumnID -> Set<Values to remove>
+  );
+
+  const processedResults = useMemo(() => {
+    const resultsForCurrentComparison = results.flat();
+    // activeComparison === allRevisionsOption
+    // ? results.flat()
+    // : results.find((result) => result[0].new_rev === activeComparison) ??
+    //   [];
+
+    const filteredResults = filterResults(
+      resultsForCurrentComparison,
+      filteringSearchTerm,
+      tableFilters,
+    );
+    return processResults(filteredResults);
+  }, [results, filteringSearchTerm, tableFilters]);
+  // }, [results, activeComparison, filteringSearchTerm, tableFilters]);
+
+  const onClearFilter = (columnId: string) => {
+    setTableFilters((oldFilters) => {
+      const newFilters = new Map(oldFilters);
+      newFilters.delete(columnId);
+      return newFilters;
+    });
+  };
+
+  const onToggleFilter = (columnId: string, filters: Set<string>) => {
+    setTableFilters((oldFilters) => {
+      const newFilters = new Map(oldFilters);
+      newFilters.set(columnId, filters);
+      return newFilters;
+    });
+  };
+
+  const styles = {
+    tableContainer: style({
+      marginTop: Spacing.Large,
+      paddingBottom: Spacing.Large,
+    }),
+  };
+
+  return (
+    <Box
+      className={styles.tableContainer}
```

Comment:
As a bit of nuance, typestyle is costly when it's used at render time -- I think its normal use should be to be called outside of components, and then the performance is OK: the style is registered once at load time, the result is then only a class name so it's pretty cheap.

But otherwise I agree; we probably don't need typestyle in general, but more because it adds yet another way of styling. Especially we should look at using `styled` more (integrated in mui) and also use MUI's theming mechanism better.

Reply:
I addressed it, I'll leave the comment open for review purposes.